modprobe: various fixes and improvements

grondo opened the pull request:
This PR started with a fix for #7011, but that fix resulted in some new issues that needed to be addressed, and so on. :disappointed: 

Finally got all the issues addressed, though there are still some corner cases in `flux module remove` that are not handled all that well. For example if you `flux modprobe remove sched feasibility` instead of `flux modprobe remove sched-fluxion-qmanager sched-fluxion-feasibility`, `sched-fluxion-resource` is not auto-unloaded since the references are not currently tracked through the module service names.

Will open an issue on that one separately.

garlick reviewed (approved):
One small nit, but o/w LGTM.
I did repeat my original test of `modprobe load resource` followed by `modprobe remove resource` and everything cleaned up properly.

grondo commented:
Thanks! Addressed the one comment and also added a patch to bash completions for the new `--dry-run` options of `flux modprobe load/remove`. I'll set MWP.
